main.go

```diff
@@ -3,81 +3,98 @@ package main
 import (
 	"context"
 	"flag"
+	"fmt"
 	"log"
-	"net"
 
 	"github.com/armon/go-socks5"
 )
 
 const (
-	allowed = true
-	denied  = false
+	allowed            = true
+	denied             = false
+	letsEncryptProd    = "acme-v02.api.letsencrypt.org"
+	letsEncryptStaging = "acme-staging-v02.api.letsencrypt.org"
 )
 
-type myRule struct {
-	addrs []net.IP
-	fqdns []string
+type allowLetsEncrypt struct{}
+
+func (m allowLetsEncrypt) Allow(ctx context.Context, req *socks5.Request) (context.Context, bool) {
+	for _, fqdn := range []string{letsEncryptProd, letsEncryptStaging} {
+		if req.DestAddr.FQDN == fqdn {
+			return ctx, allowed
+		}
+	}
+	return ctx, denied
+}
+
+type logAll struct {
```

Comment:
Added something like a "logging middleware" so we could enable/disable logging.  In the old version we always logged, this new setup allows us to turn it off if we want.  (Note that logging was very helpful in development)